packages/twenty-server/src/engine/metadata-modules/search-field-metadata/search-field-metadata.service.ts

```diff
@@ -0,0 +1,89 @@
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+
+import { Repository } from 'typeorm';
+
+import { SearchFieldMetadataEntity } from 'src/engine/metadata-modules/search-field-metadata/search-field-metadata.entity';
+import { type FieldTypeAndNameMetadata } from 'src/engine/workspace-manager/workspace-sync-metadata/utils/get-ts-vector-column-expression.util';
+import {
+  isSearchableFieldType,
+  type SearchableFieldType,
+} from 'src/engine/workspace-manager/workspace-sync-metadata/utils/is-searchable-field.util';
+
+@Injectable()
+export class SearchFieldMetadataService {
+  constructor(
+    @InjectRepository(SearchFieldMetadataEntity)
+    private readonly searchFieldMetadataRepository: Repository<SearchFieldMetadataEntity>,
+  ) {}
+
+  async getSearchableFields(
+    objectMetadataId: string,
+  ): Promise<FieldTypeAndNameMetadata[]> {
+    const searchFieldMetadata = await this.searchFieldMetadataRepository.find({
+      where: { objectMetadataId },
+      relations: ['fieldMetadata'],
+    });
+
+    return searchFieldMetadata
+      .filter((searchField) =>
+        isSearchableFieldType(searchField.fieldMetadata.type),
+      )
+      .map((searchField) => ({
+        name: searchField.fieldMetadata.name,
+        type: searchField.fieldMetadata.type as SearchableFieldType,
+      }));
+  }
+
+  async isFieldSearchable(
```

Comment:
It's hard to know yet since this isn't used anywhere but let's be extra careful with N+1 query issues when implementing this. I feel like this might be an issue with your current methods